src/compiler/scala/tools/nsc/reporters/Reporter.scala

```diff
@@ -97,11 +93,16 @@ object Reporter {
 abstract class FilteringReporter extends Reporter {
   def settings: Settings
 
+  @deprecatedOverriding("override the `doReport` overload instead", "2.13.12")
+  @deprecated("use the `doReport` overload instead", "2.13.12")
+  def doReport(pos: Position, msg: String, severity: Severity): Unit = doReport(pos, msg, severity, Nil)
```

Comment:
I see that this is kept, so we should be asafe, I think doReport is used in scalameta/scalamete and scalameta/mdoc.

Reply:
Calling the method is source compatible, but extending `FilteringReporter` is not. The abstract method used to be

```scala
  def doReport(pos: Position, msg: String, severity: Severity): Unit
```

and now it's

```scala
  @deprecatedOverriding("override the `doReport` overload instead", "2.13.12")
  @deprecated("use the `doReport` overload instead", "2.13.12")
  def doReport(pos: Position, msg: String, severity: Severity): Unit = doReport(pos, msg, severity, Nil)

  override def doReport(pos: Position, msg: String, severity: Severity, actions: List[CodeAction]): Unit
```

Reply:
That will be a problem for mdoc then since we don't publish for each Scala version :/ This of course causes us problems, but on the other hand we can support all the Scala versions without separate publishing 

Any idea how we should work around that? I can take a look at it after Scala Days, but I am bit terrified as I don't fully remember why we wrote it this way. I guess some of the classes were not available in the older versions and I had to work around it this way.

Reply:
OK, so ideally an mdoc release should be binary compatible with all Scala 2.13 versions, i.e.,
  - existing ones should continue to work with new Scala releases, and
  - new mdoc releases built with a new Scala version should work with older Scala versions?

Reply:
That's the idea, but it's based on an assumption that the Report API wouldn't change too much :sweat: which is clearly a wrong assumption.

Reply:
I pushed a small change that should restore binary compatibility (both ways) with existing FilteringReporter subclasses. I tested it with the following:

```scala
import scala.tools.nsc.Settings
import scala.reflect.internal.util.Position
// import scala.reflect.internal.util.CodeAction
import scala.reflect.internal.Reporter.Severity
import scala.annotation.nowarn
import scala.tools.nsc.reporters.FilteringReporter

trait MyRepT extends FilteringReporter { self: MyRep =>
  @nowarn("cat=deprecation")
  override def doReport(pos: Position, msg: String, severity: Severity): Unit =
    add(pos, msg, severity)

  // override def doReport(pos: Position, msg: String, severity: Severity, actions: List[CodeAction]): Unit =
    // add(pos, msg, severity)
}

class MyRep(val settings: Settings) extends MyRepT {
  def add(pos: Position, msg: String, severity: Severity): Unit =
    println(s"[myrep] $msg")
}
```

Compiling this class with 2.13.10, I can use the reporter with both 2.13.10 and this PR

```
➜ sandbox java -cp $(cs fetch -p org.scala-lang:scala-compiler:2.13.10):. scala.tools.nsc.Main -usejavacp -Xreporter:MyRep A.scala
[myrep] type mismatch;
 found   : String("")
 required: Int

➜ sandbox qsc -toolcp . -Xreporter:MyRep A.scala
[myrep] type mismatch;
 found   : String("")
 required: Int
```

I also compiled the class with this PR's compiler and it works with 2.13.10, with or without un-commenting the `doReport` overload.

Reply:
Thanks! I think that should work for us! I can confirm 100% when there is a nightly version released